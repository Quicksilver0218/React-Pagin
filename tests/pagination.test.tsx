/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import Pagination from "../src";

test("Hide Ellipsis", () => {
  expect(
    render(
      <Pagination
        currentPage={1}
        pageCount={11}
        edgeItemCount={1}
        middleItemRange={3}
        firstElement="first"
        lastElement="last"
        previousElement="prev"
        nextElement="next"
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={1}
      />
    ).container
  ).toMatchSnapshot();

  expect(
    render(
      <Pagination
        currentPage={7}
        pageCount={13}
        edgeItemCount={1}
        middleItemRange={3}
        firstElement="first"
        lastElement="last"
        previousElement="prev"
        nextElement="next"
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={2}
      />
    ).container
  ).toMatchSnapshot();
});

test("1 Ellipsis", () => {
  expect(
    render(
      <Pagination
        currentPage={1}
        pageCount={12}
        edgeItemCount={1}
        middleItemRange={3}
        firstElement="first"
        lastElement="last"
        previousElement="prev"
        nextElement="next"
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={1}
      />
    ).container
  ).toMatchSnapshot();

  expect(
    render(
      <Pagination
        currentPage={6}
        pageCount={12}
        edgeItemCount={1}
        middleItemRange={3}
        firstElement="first"
        lastElement="last"
        previousElement="prev"
        nextElement="next"
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={1}
      />
    ).container
  ).toMatchSnapshot();

  expect(
    render(
      <Pagination
        currentPage={7}
        pageCount={12}
        edgeItemCount={1}
        middleItemRange={3}
        firstElement="first"
        lastElement="last"
        previousElement="prev"
        nextElement="next"
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={1}
      />
    ).container
  ).toMatchSnapshot();

  expect(
    render(
      <Pagination
        currentPage={12}
        pageCount={12}
        edgeItemCount={1}
        middleItemRange={3}
        firstElement="first"
        lastElement="last"
        previousElement="prev"
        nextElement="next"
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={1}
      />
    ).container
  ).toMatchSnapshot();
});

test("2 Ellipsis", () => {
  expect(
    render(
      <Pagination
        currentPage={7}
        pageCount={13}
        edgeItemCount={1}
        middleItemRange={3}
        firstElement="first"
        lastElement="last"
        previousElement="prev"
        nextElement="next"
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={1}
      />
    ).container
  ).toMatchSnapshot();
});

test("All Default", () => {
  expect(
    render(
      <Pagination
        currentPage={7}
        pageCount={13}
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
      />
    ).container
  ).toMatchSnapshot();

  expect(
    render(
      <Pagination
        currentPage={5}
        pageCount={10}
        pageElement={({page}) => <>{page}</>}
      />
    ).container
  ).toMatchSnapshot();
});

test("Default Ellipsis Size", () => {
  expect(
    render(
      <Pagination
        currentPage={1}
        pageCount={12}
        edgeItemCount={1}
        middleItemRange={3}
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
      />
    ).container
  ).toMatchSnapshot();

  expect(
    render(
      <Pagination
        currentPage={6}
        pageCount={11}
        edgeItemCount={1}
        middleItemRange={3}
        pageElement={({page}) => <>{page}</>}
      />
    ).container
  ).toMatchSnapshot();
});

test("Middle Item Range", () => {
  expect(
    render(
      <Pagination
        currentPage={6}
        pageCount={11}
        edgeItemCount={1}
        middleItemRange={2}
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={1}
      />
    ).container
  ).toMatchSnapshot();

  expect(
    render(
      <Pagination
        currentPage={5}
        pageCount={10}
        edgeItemCount={1}
        middleItemRange={2}
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={1}
      />
    ).container
  ).toMatchSnapshot();

  expect(
    render(
      <Pagination
        currentPage={6}
        pageCount={10}
        edgeItemCount={1}
        middleItemRange={2}
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={1}
      />
    ).container
  ).toMatchSnapshot();
});

test("Ellipsis Size", () => {
  expect(
    render(
      <Pagination
        currentPage={6}
        pageCount={11}
        edgeItemCount={1}
        middleItemRange={2}
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={2}
      />
    ).container
  ).toMatchSnapshot();

  expect(
    render(
      <Pagination
        currentPage={6}
        pageCount={12}
        edgeItemCount={1}
        middleItemRange={2}
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={2}
      />
    ).container
  ).toMatchSnapshot();

  expect(
    render(
      <Pagination
        currentPage={7}
        pageCount={12}
        edgeItemCount={1}
        middleItemRange={2}
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={2}
      />
    ).container
  ).toMatchSnapshot();
});