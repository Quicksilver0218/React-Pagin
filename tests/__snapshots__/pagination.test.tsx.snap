// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 Ellipsis 1`] = `
<div>
  first
  prev
  1
  2
  3
  4
  5
  6
  7
  8
  9
  end
  12
  next
  last
</div>
`;

exports[`1 Ellipsis 2`] = `
<div>
  first
  prev
  1
  2
  3
  4
  5
  6
  7
  8
  9
  end
  12
  next
  last
</div>
`;

exports[`1 Ellipsis 3`] = `
<div>
  first
  prev
  1
  start
  4
  5
  6
  7
  8
  9
  10
  11
  12
  next
  last
</div>
`;

exports[`1 Ellipsis 4`] = `
<div>
  first
  prev
  1
  start
  4
  5
  6
  7
  8
  9
  10
  11
  12
  next
  last
</div>
`;

exports[`2 Ellipsis 1`] = `
<div>
  first
  prev
  1
  start
  4
  5
  6
  7
  8
  9
  10
  end
  13
  next
  last
</div>
`;

exports[`All Default 1`] = `
<div>
  1
  start
  4
  5
  6
  7
  8
  9
  10
  end
  13
</div>
`;

exports[`All Default 2`] = `
<div>
  1
  2
  3
  4
  5
  6
  7
  8
  10
</div>
`;

exports[`Default Ellipsis Size 1`] = `
<div>
  1
  2
  3
  4
  5
  6
  7
  8
  9
  end
  12
</div>
`;

exports[`Default Ellipsis Size 2`] = `
<div>
  1
  3
  4
  5
  6
  7
  8
  9
  11
</div>
`;

exports[`Ellipsis Size 1`] = `
<div>
  1
  2
  3
  4
  5
  6
  7
  8
  9
  10
  11
</div>
`;

exports[`Ellipsis Size 2`] = `
<div>
  1
  2
  3
  4
  5
  6
  7
  8
  end
  12
</div>
`;

exports[`Ellipsis Size 3`] = `
<div>
  1
  start
  5
  6
  7
  8
  9
  10
  11
  12
</div>
`;

exports[`Hide Ellipsis 1`] = `
<div>
  first
  prev
  1
  2
  3
  4
  5
  6
  7
  8
  9
  10
  11
  next
  last
</div>
`;

exports[`Hide Ellipsis 2`] = `
<div>
  first
  prev
  1
  2
  3
  4
  5
  6
  7
  8
  9
  10
  11
  12
  13
  next
  last
</div>
`;

exports[`Middle Item Range 1`] = `
<div>
  1
  start
  4
  5
  6
  7
  8
  end
  11
</div>
`;

exports[`Middle Item Range 2`] = `
<div>
  1
  2
  3
  4
  5
  6
  7
  end
  10
</div>
`;

exports[`Middle Item Range 3`] = `
<div>
  1
  start
  4
  5
  6
  7
  8
  9
  10
</div>
`;
